uportal-war/src/main/java/org/jasig/portal/utils/LdapQueryUtils.java

```diff
@@ -0,0 +1,34 @@
+package org.jasig.portal.utils;
+
+public class LdapQueryUtils {
+
+	/**
+	 * 
+	 * @param query
+	 * @return
+	 */
+	public static String escapeSpecialCharacters(String query) {
+        // We need to escape regex special characters that appear in the query string...
+        final String[][] specials = new String[][] {
+                            /* backslash must come first! */
+                            new String[] { "\\", "\\\\"}, 
+                            new String[] { "[", "\\[" }, 
+                            /* closing ']' isn't needed b/c it's a normal character w/o a preceding '[' */
+                            new String[] { "{", "\\{" }, 
+                            /* closing '}' isn't needed b/c it's a normal character w/o a preceding '{' */
+                            new String[] { "^", "\\^" },
+                            new String[] { "$", "\\$" },
+                            new String[] { ".", "\\." },
+                            new String[] { "|", "\\|" },
+                            new String[] { "?", "\\?" },
+                            new String[] { "*", "\\*" },
+                            new String[] { "+", "\\+" },
+                            new String[] { "(", "\\(" },
+                            new String[] { ")", "\\)" }
+                        };
```

Comment:
Correction to above.  You should escape the characters specified above using \ (in Java Strings \\) plus 2 digit code which is what the spec says and is also what I've found on various other Google searches. Do not escape with just a backslash.  That is appropriate for the regex escaping but not an LDAP Search Filter escaping.